lib/ui/workoutPlayer/workout_player.dart

```diff
@@ -0,0 +1,234 @@
+  return Stack(
+    children: <Widget> [
+      Container(
+        color: Colors.black,
+        alignment: Alignment.center,
+        width: MediaQuery.of(context).size.width,
+        height: MediaQuery.of(context).size.height * 0.75,
+        child: AspectRatio(
+          aspectRatio: 16/9,
+          child: VideoPlayer(videoSrcUrl, UniqueKey(), handleVideoCompleted)
+        ),
+      ),
+      countDownTimer(context)
+    ]
+  );
+}
+
+Widget countDownTimer(BuildContext context) {
+  return Positioned(
+    top: 0,
+    left: MediaQuery.of(context).size.width / 30,
+    child: CircularCountDownTimer(
+      duration: 30,
+      initialDuration: 0,
+      width: MediaQuery.of(context).size.width / 30,
+      height: 200,
+      ringColor: Colors.grey,
+      fillColor: Colors.white,
+      strokeWidth: 3.0,
+      strokeCap: StrokeCap.round,
+      textStyle: const TextStyle(
+        fontSize: 20.0,
+        color: Colors.white,
+        fontWeight: FontWeight.bold
+      ),
+      textFormat: CountdownTextFormat.S,
+      isReverse: true,
+      isTimerTextShown: true,
+      autoStart: true,
+    )
+  );
+}
+
+Widget videoDescription() {
+  return const Text('This is the workout description', textAlign: TextAlign.left,);
+}
+
+Widget workoutTitle() {
+  return Container(
+    alignment: Alignment.centerLeft,
+    child: const Text(
+      'Title of Workout',
+      textAlign: TextAlign.left,
+      style: TextStyle(
+        fontWeight: FontWeight.bold,
+        color: Colors.black,
+        fontSize: 30
+      )
+    )
+  );
+}
+
+Widget exerciseList(BuildContext context, List<Exercise> exercises, Function setCurExercise, ItemScrollController _itemScrollController, curExerciseIndex) {
+  return exercises.isNotEmpty
+    ? ScrollablePositionedList.builder(
+        itemScrollController: _itemScrollController,
+        itemCount: exercises.length,
+        itemBuilder: (BuildContext context, int index) {
+          var itemKey = GlobalKey();
+          return GestureDetector(
+            key: itemKey,
+            child: exercise(context, exercises[index], curExerciseIndex == index),
+            onTap: () => {
+              setCurExercise(index),
+              playExercise(itemKey, index, _itemScrollController),
+            }
+          );
+        }
+      )
+      : Row(
+        children: const [
+          CircularProgressIndicator(),
+          Text("Loading")
+        ]
+      );
+}
+
+Future playExercise(itemKey, index, _scrollController) async {
+  await _scrollController.scrollTo(
+    index: index,
+    duration: Duration(milliseconds: 500),
+    curve: Curves.easeInOutCubic
+  );
+}
+
+Widget exercise(BuildContext context, Exercise exercise, bool isCurIndex) {
+  Duration duration = Duration(seconds: exercise.length);
+  String sDuration = "${duration.inMinutes.toString().padLeft(2, "0")}:${duration.inSeconds.remainder(60).toString().padLeft(2, "0")}";
```

Comment:
After making the change to the Exercise model, we can directly use the duration here